### PR TITLE
[Service Connector] `az webapp connection`: Add new parameter `--slot` to support webapp slot connection

### PR DESCRIPTION
**Pull request by LianwMS**
**Related command**

az webapp connection delete/create/show/list/list-configuration/validate/update

**Description**

Add service connector support for web app slot

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py [hunk 1]
```diff
@@ -83,6 +84,16 @@ def add_source_resource_block(context, source, enable_id=True, validate_source_i
...
+    # slot parameter
+    if source == RESOURCE.WebApp:
+        for arg, content in SOURCE_RESOURCES_OPTIONAL_PARAMS.get(source).items():
+            context.argument(arg, options_list=content.get(
+                'options'), type=str, help=content.get('help'))
+    else: 
+        for resource, args in SOURCE_RESOURCES_OPTIONAL_PARAMS.items():
+            if resource != source:
+                for arg in args:
+                    context.ignore(arg)
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py:96]:
```suggestion
    # slot parameter
    for resource, args in SOURCE_RESOURCES_OPTIONAL_PARAMS.items():
        for arg, content in args.items():
            if source == resource:
                context.argument(arg, ...)
            else:
                context.ignore(arg)
```

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py [hunk 2]
```diff
@@ -270,6 +270,16 @@ class CLIENT_TYPE(Enum):
     },
 }
 
+# The dict defines the optional parameters used in the source resources for WebApp creation.
+SOURCE_RESOURCES_OPTIONAL_PARAMS = {
+    RESOURCE.WebApp: {
+        'slot': {
+            'options': ['--slot'],
+            'help': 'The name of the slot. Default to the productions slot if not specified.',
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:278]: `productions` --> `production`?

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py (hunk 1 above)
Comment (houk-ms) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py:96]: could be more concise as suggested

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py [hunk 3]
```diff
@@ -622,7 +629,20 @@ def apply_source_args(cmd, namespace, arg_values):
...
         )
+    apply_source_optional_args(cmd, namespace, arg_values)
 
+def apply_source_optional_args(cmd, namespace, arg_values):
+    '''Set source resource id by optional arg_values
+    '''
+    source = get_source_resource_name(cmd)
+    if source == RESOURCE.WebApp:
+        if arg_values.get('slot', None):
+            resource = '/subscriptions/{subscription}/resourceGroups/{source_resource_group}/providers/Microsoft.Web/sites/{site}/slots/{slot}'
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:640]: better reuse
```suggestion
            resource = '{}/slots/{slot}'.format(SOURCE_RESOURCES.get(RESOURCE.WebApp))
```

**Review by houk-ms — COMMENTED** [review]
1. run the commands below to check and resolve CI failures locally
```
azdev test serviceconnector
azdev style serviceconnector 
azdev linter serviceconnector 
```

2. Better to add a testcase for slot, and generate a recording by
```
azdev test <yourtestcasename> --live
```

3. PR title needs follow CLI PR naming format, a suggestion here
```
[Service Connector] `az webapp connection`: Add new parameter `--slot` to support webapp slot connection
```

**Review by houk-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**